### PR TITLE
Reorganize directory structure

### PR DESCRIPTION
**Pull request by aaronmars**
The old directory structure contained folders for `src`, `stories`, and `tests`. This has been replaced by a single source folder that contains sub-directories for each component.
Inside each of these, sub-directories, related components are grouped together along with their tests and stories.
This lays the groundwork for a future where every exported component has a story, and has tests associated with it.

While the main goal of this PR is to establish this structure, additional work was done to remove the library bundling and only output cjs, esm and types.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -9,9 +9,7 @@ const packageJson = require('./package.json');
  */
```
Comment (JeremyRH) [inline webpack.config.js:9]: Do we still need webpack?

Comment (aaronmars) [reply webpack.config.js:9]: Oh man... Of all the things to forget.

**Review by JeremyRH — APPROVED** [review]

File: babel.esm.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  extends: './babel.build.config.js',
+  presets: [
+    // Dont transform modules for the esm build
+    ['@babel/preset-env', { modules: false }],
+  ],
+};
```
Comment (steventhan) [inline babel.esm.config.js:7]: Just curious, does `esm` build not transpile typescript and jsx?

Comment (JeremyRH) [reply babel.esm.config.js:7]: It does. The module syntax `import` is not transformed.

File: scripts/copyDTSFiles.js [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
+/* eslint-disable import/no-extraneous-dependencies, no-console */
+const { copy } = require('fs-extra');
 const glob = require('glob');
-const fse = require('fs-extra');
+const path = require('path');
 
-// copies all .d.ts files to lib/ directory
-// inspired by https://github.com/mui-org/material-ui/blob/v1-beta/scripts/copy-files.js
+// copies all .d.ts files to esm/ directory
+// This can go away once there are no explicit
```
Comment (steventhan) [inline scripts/copyDTSFiles.js:9]: Not sure if this comment is complete 🤔

**Comment by JeremyRH** [comment]
Top level `util` directory is still named util?

File: scripts/copyDTSFiles.js (hunk 3 above)
Comment (aaronmars) [reply scripts/copyDTSFiles.js:9]: It is not complete! Fixing now.
The intent is that this operation can go away when everything is converted to TS and we have no more explicit .d.ts files.

**Comment by aaronmars** [comment]
> Top level `util` directory is still named util?

@JeremyRH : Totally forgot that change. Doing it now.

**Review by JeremyRH — APPROVED** [review]

**Pull request merged**